web_client/panels/AnnotationSelector.js

```diff
@@ -33,21 +72,39 @@ var AnnotationSelector = Panel.extend({
         this.collection.offset = 0;
         this.collection.fetch({itemId: item.id})
             .then(() => this.render());
+        return this;
     },
-    render() {
-        this.$el.html(annotationSelectorWidget({
-            annotations: this.collection.toArray(),
-            id: 'annotation-panel-container',
-            title: 'Annotations'
-        }));
-        this.$('.s-panel-content').collapse({toggle: false});
+
+    /**
+     * Set the image "viewer" instance.  This should be a subclass
+     * of `large_image/imageViewerWidget` that is capable of rendering
+     * annotations.
+     */
+    setViewer(viewer) {
+        this.viewer = viewer;
         return this;
     },
+
+    /**
+     * Toggle the renderering of a specific annotation.  Sets the
+     * `displayed` attribute of the `AnnotationModel`.
+     */
     toggleAnnotation(evt) {
-        var id = $(evt.currentTarget).data('id');
+        var id = $(evt.currentTarget).parent('.h-annotation').data('id');
         var model = this.collection.get(id);
         model.set('displayed', !model.get('displayed'));
-        this.render();
+    },
+
+    /**
+     * Delete an annotation from the server.
+     */
+    deleteAnnotation(evt) {
```

Comment:
When I delete a huge annotation (one with millions of elements), the delete takes a significant amount of time (more than a minute).  While that should be improved as a separate issue, there is no feedback that the delete has been triggered.  Perhaps the annotation entry have a class added to it that disables the associated buttons and greys it out (a reload or update might revert this look, but at least there would be some feedback).